pyrokinetics/gk_code/GKInputCGYRO.py

```diff
@@ -36,23 +38,71 @@ class GKInputCGYRO(GKInput):
         "kappa": "KAPPA",
         "s_kappa": "S_KAPPA",
         "delta": "DELTA",
+        "shat": "S",
+        "shift": "SHIFT",
+    }
+
+    pyro_cgyro_mxh = {
+        "rho": "RMIN",
+        "Rmaj": "RMAJ",
+        "Z0": "ZMAG",
+        "dZ0dr": "DZMAG",
+        "q": "Q",
+        "kappa": "KAPPA",
+        "s_kappa": "S_KAPPA",
+        "delta": "DELTA",
+        "s_delta": "DELTA",
         "zeta": "ZETA",
-        "s_zeta": "S_ZETA",
+        "s_zeta": "ZETA",
         "shat": "S",
         "shift": "SHIFT",
+        "cn0": "SHAPE_COS0",
+        "cn1": "SHAPE_COS1",
+        "cn2": "SHAPE_COS2",
+        "cn3": "SHAPE_COS3",
+        "sn3": "SHAPE_SIN3",
+        "dcndr0": "SHAPE_S_COS0",
+        "dcndr1": "SHAPE_S_COS1",
+        "dcndr2": "SHAPE_S_COS2",
+        "dcndr3": "SHAPE_S_COS3",
+        "dsndr3": "SHAPE_S_SIN3",
     }
 
-    pyro_cgyro_miller_defaults = {
+    pyro_cgyro_mxh_defaults = {
         "rho": 0.5,
         "Rmaj": 3.0,
+        "Z0": 0.0,
+        "dZ0dr": 0.0,
         "q": 2.0,
         "kappa": 1.0,
         "s_kappa": 0.0,
         "delta": 0.0,
+        "s_delta": 0.0,
         "zeta": 0.0,
         "s_zeta": 0.0,
         "shat": 1.0,
         "shift": 0.0,
+        "cn0": 0.0,
+        "cn1": 0.0,
+        "cn2": 0.0,
+        "cn3": 0.0,
+        "sn3": 0.0,
+        "dcndr0": 0.0,
+        "dcndr1": 0.0,
+        "dcndr2": 0.0,
+        "dcndr3": 0.0,
+        "dsndr3": 0.0,
+    }
+
+    pyro_cgyro_miller_defaults = {
+        "rho": 0.5,
+        "Rmaj": 3.0,
+        "q": 2.0,
+        "kappa": 1.0,
+        "s_kappa": 0.0,
+        "delta": 0.0,
```

Comment:
no s_delta?

Reply:
Also given the mxh ones expand this set can you remove the duplication of rho, rmaj etc defaults by defining the base first and then making mxh start with a copy of this which just adds the extra values?

Reply:
There is but its not a one to one mapping from the gk_code to the pyro definition. (CGYRO is different to the Miller `s_delta` by a factor `1/sqrt(1-delta^2)`, which is what this list is for so you have to manually set it. see below

https://github.com/pyro-kinetics/pyrokinetics/blob/bcc6843e06fc6cb08c441b907dded6d6e6754986/pyrokinetics/gk_code/GKInputCGYRO.py#L264-L266